### ISSUE TITLE
Allow to specify to use `undefined` or `void`

### ISSUE THREAD

**Issue by garyking** [gajus/eslint-plugin-jsdoc#399]
When a function returns nothing, we can use the return type of `undefined` or `void`. Please create a new rule to allow us to choose one or the other, and make the other one invalid (therefore reporting an error on it).

**Comment by brettz9** [comment]
You can use https://github.com/gajus/eslint-plugin-jsdoc#settings-to-configure-check-types-and-no-undefined-types

Namely in your config:

```js
    settings: {
        jsdoc: {
          preferredTypes: {
            undefined: 'void',
          },
        },
      },
```

...to recommend use of `void` over `undefined`.

Or use `void: 'undefined'` to do the opposite.

You can also use `message` and `replacement` to give a reason:

```js
    settings: {
        jsdoc: {
          preferredTypes: {
            undefined: {
                // Produces, e.g., "You shouldn't use `undefined` on the tag @param with the name "foo"; we want to be clear that we are not really returning a value to be handled."
                message: 'You shouldn\'t use `undefined` on the tag @{{tagName}} with the name {{tagValue}}; we want to be clear that we are not really returning a value to be handled.',
                replacement: 'void'
            }
          },
        },
      },
```
